modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -1595,6 +1598,11 @@ CompletableFuture<Void> saveDataNodesToMetaStorageOnScaleDown(int zoneId, long r
                         .thenApply(StatementResult::getAsBoolean)
                         .thenCompose(invokeResult -> inBusyLock(busyLock, () -> {
                             if (invokeResult) {
+                                if (newDataNodes.isEmpty()) {
```

Comment:
How it's supposed to work?
Let's say that we've prepared new dataNodesAndTriggerKeyUpd with new revision and **same** newDataNodesBytes.
Are we going to actually update dataNodes with same value? Are we going to consider such same to same change as new rebalance trigger?

Reply:
1. Same to same a data nodes change will not trigger a new rebalance because RebalanceUtil#updatePendingAssignmentsKeys filters the same assignments.
2. I think we can not save the same data nodes, but it will require more changes in the code. Actually a meta storage invoke with the same data nodes can not complete successfully and `saveDataNodesToMetaStorageOnScaleDown` can be repeated again but with a different set of nodes.